compiler/rustc_session/src/session.rs

```diff
@@ -252,7 +252,8 @@ impl Session {
 
     pub fn local_crate_source_file(&self) -> Option<PathBuf> {
         let path = self.io.input.opt_path()?;
-        if self.should_prefer_remapped_for_codegen() {
+        // FIXME: The remap path scope should probably not be hardcoded.
+        if self.should_prefer_remapped(RemapPathScopeComponents::DEBUGINFO) {
```

Comment:
Let's fix the FIXME 🙂 The `local_crate_source_file` method is called in at least one DIAGNOSTIC context.

Reply:
You're right. I've added a new commit switching fixing it.